index.js

```diff
@@ -1,14 +1,29 @@
 'use strict';
+
 require('harmony-reflect');
 
-module.exports = function (obj) {
-	return new Proxy(obj, {
+module.exports = function (target, readonly) {
```

Comment:
I think that we should keep the current API and add the decorator on top, either by detecting if `target` is a constructor or an object or adding a .decorator() method.